core/imds/src/main/java/software/amazon/awssdk/imds/internal/DefaultEc2MetadataAsyncClient.java

```diff
@@ -84,6 +96,21 @@ public static Ec2MetadataAsyncClient.Builder builder() {
         return new DefaultEc2MetadataAsyncClient.Ec2MetadataAsyncBuilder();
     }
 
+    private Supplier<CompletableFuture<String>> createCachedTokenSupplier(TokenCacheStrategy tokenCacheStrategy) {
+        SdkHttpFullRequest baseTokenRequest = requestMarshaller.createTokenRequest(tokenTtl);
+        if (tokenCacheStrategy == TokenCacheStrategy.NONE) {
+            return () -> sendAsyncRequest(baseTokenRequest);
+        }
+        return CachedSupplier
+            .builder(() -> {
+                CompletableFuture<String> tokenFuture = sendAsyncRequest(baseTokenRequest);
+                return RefreshResult.builder(tokenFuture)
+                                    .staleTime(Instant.now().plus(tokenTtl))
```

Comment:
It seems the timer starts before the request finishes. Token request could take up to 1 second in rare cases, and it could cause issues if the token ttl is lower than 1 second